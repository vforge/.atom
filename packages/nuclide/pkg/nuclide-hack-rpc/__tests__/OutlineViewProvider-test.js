"use strict";

function _OutlineView() {
  const data = require("../lib/OutlineView");

  _OutlineView = function () {
    return data;
  };

  return data;
}

/**
 * Copyright (c) 2015-present, Facebook, Inc.
 * All rights reserved.
 *
 * This source code is licensed under the license found in the LICENSE file in
 * the root directory of this source tree.
 *
 *  strict-local
 * @format
 * @emails oncall+nuclide
 */
describe('outlineFromHackIdeOutline', () => {
  it('function', () => {
    const actualOutline = (0, _OutlineView().outlineFromHackIdeOutline)([{
      name: 'f',
      representativeName: 'f',
      modifiers: [],
      span: {
        line_start: 3,
        line_end: 4,
        char_start: 1,
        char_end: 1,
        filename: ''
      },
      kind: 'function',
      params: [{
        kind: 'param',
        position: {
          filename: '',
          char_end: 17,
          char_start: 16,
          line: 3
        },
        modifiers: [],
        span: {
          line_start: 3,
          line_end: 3,
          char_start: 12,
          char_end: 17,
          filename: ''
        },
        name: '$x'
      }, {
        kind: 'param',
        position: {
          filename: '',
          char_end: 21,
          char_start: 20,
          line: 3
        },
        modifiers: [],
        span: {
          line_start: 3,
          line_end: 3,
          char_start: 20,
          char_end: 21,
          filename: ''
        },
        name: '$y'
      }],
      position: {
        filename: '',
        char_end: 10,
        line: 3,
        char_start: 10
      }
    }]);
    expect(actualOutline).toMatchSnapshot();
  });
  it('class', () => {
    const actualOutline = (0, _OutlineView().outlineFromHackIdeOutline)([{
      name: 'D',
      children: [{
        position: {
          filename: '',
          char_end: 14,
          char_start: 14,
          line: 11
        },
        kind: 'typeconst',
        name: 'Y',
        modifiers: [],
        span: {
          char_end: 20,
          filename: '',
          line_end: 11,
          char_start: 3,
          line_start: 11
        }
      }, {
        kind: 'property',
        position: {
          char_start: 39,
          line: 12,
          char_end: 40,
          filename: ''
        },
        modifiers: ['private'],
        span: {
          line_end: 12,
          char_start: 39,
          char_end: 40,
          filename: '',
          line_start: 12
        },
        name: 'x'
      }, {
        name: '__construct',
        modifiers: ['public'],
        span: {
          char_end: 3,
          filename: '',
          line_end: 14,
          char_start: 3,
          line_start: 12
        },
        position: {
          char_start: 19,
          line: 12,
          filename: '',
          char_end: 29
        },
        kind: 'method',
        params: [{
          position: {
            filename: '',
            char_end: 40,
            char_start: 39,
            line: 12
          },
          kind: 'param',
          name: '$x',
          modifiers: ['private'],
          span: {
            line_start: 12,
            char_start: 39,
            line_end: 12,
            filename: '',
            char_end: 40
          }
        }]
      }, {
        modifiers: ['abstract'],
        span: {
          line_start: 15,
          char_end: 18,
          filename: '',
          line_end: 15,
          char_start: 18
        },
        name: 'Z',
        kind: 'const',
        position: {
          filename: '',
          char_end: 18,
          line: 15,
          char_start: 18
        }
      }, {
        kind: 'const',
        position: {
          char_end: 9,
          filename: '',
          char_start: 9,
          line: 16
        },
        name: 'X',
        modifiers: [],
        span: {
          char_start: 9,
          line_end: 16,
          filename: '',
          char_end: 14,
          line_start: 16
        }
      }],
      span: {
        filename: '',
        char_end: 1,
        char_start: 1,
        line_end: 17,
        line_start: 10
      },
      modifiers: ['abstract'],
      kind: 'class',
      position: {
        char_start: 16,
        line: 10,
        filename: '',
        char_end: 16
      }
    }]);
    expect(actualOutline).toMatchSnapshot();
  });
  it('trait', () => {
    const actualOutline = (0, _OutlineView().outlineFromHackIdeOutline)([{
      position: {
        filename: '',
        char_end: 7,
        char_start: 7,
        line: 19
      },
      kind: 'trait',
      children: [{
        name: 'f',
        span: {
          char_start: 3,
          line_end: 22,
          filename: '',
          char_end: 3,
          line_start: 20
        },
        modifiers: ['public'],
        position: {
          char_start: 19,
          line: 20,
          filename: '',
          char_end: 19
        },
        kind: 'method',
        params: []
      }],
      name: 'T',
      modifiers: [],
      span: {
        char_start: 1,
        line_end: 23,
        filename: '',
        char_end: 1,
        line_start: 19
      }
    }]);
    expect(actualOutline).toMatchSnapshot();
  });
  it('interface', () => {
    const actualOutline = (0, _OutlineView().outlineFromHackIdeOutline)([{
      kind: 'interface',
      position: {
        line: 25,
        char_start: 11,
        filename: '',
        char_end: 11
      },
      span: {
        filename: '',
        char_end: 14,
        char_start: 1,
        line_end: 25,
        line_start: 25
      },
      modifiers: [],
      name: 'I',
      children: []
    }]);
    expect(actualOutline).toMatchSnapshot();
  });
  it('enum', () => {
    const actualOutline = (0, _OutlineView().outlineFromHackIdeOutline)([{
      name: 'En',
      children: [{
        kind: 'const',
        position: {
          line: 28,
          char_start: 3,
          filename: '',
          char_end: 3
        },
        span: {
          filename: '',
          char_end: 7,
          char_start: 3,
          line_end: 28,
          line_start: 28
        },
        modifiers: [],
        name: 'C'
      }],
      span: {
        line_start: 27,
        line_end: 29,
        char_start: 1,
        char_end: 1,
        filename: ''
      },
      modifiers: [],
      position: {
        char_end: 7,
        filename: '',
        char_start: 6,
        line: 27
      },
      kind: 'enum'
    }]);
    expect(actualOutline).toMatchSnapshot();
  });
});