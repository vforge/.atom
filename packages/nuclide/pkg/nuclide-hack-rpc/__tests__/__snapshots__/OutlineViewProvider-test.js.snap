// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`outlineFromHackIdeOutline class 1`] = `
Object {
  "outlineTrees": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [],
          "endPosition": Point {
            "column": 19,
            "row": 10,
          },
          "representativeName": "Y",
          "startPosition": Point {
            "column": 13,
            "row": 10,
          },
          "tokenizedText": Array [
            Object {
              "kind": "keyword",
              "value": "const",
            },
            Object {
              "kind": "whitespace",
              "value": " ",
            },
            Object {
              "kind": "keyword",
              "value": "type",
            },
            Object {
              "kind": "whitespace",
              "value": " ",
            },
            Object {
              "kind": "class-name",
              "value": "Y",
            },
          ],
        },
        Object {
          "children": Array [],
          "endPosition": Point {
            "column": 39,
            "row": 11,
          },
          "representativeName": "x",
          "startPosition": Point {
            "column": 38,
            "row": 11,
          },
          "tokenizedText": Array [
            Object {
              "kind": "keyword",
              "value": "private",
            },
            Object {
              "kind": "whitespace",
              "value": " ",
            },
            Object {
              "kind": "keyword",
              "value": "property",
            },
            Object {
              "kind": "whitespace",
              "value": " ",
            },
            Object {
              "kind": "method",
              "value": "x",
            },
          ],
        },
        Object {
          "children": Array [],
          "endPosition": Point {
            "column": 2,
            "row": 13,
          },
          "representativeName": "__construct",
          "startPosition": Point {
            "column": 18,
            "row": 11,
          },
          "tokenizedText": Array [
            Object {
              "kind": "keyword",
              "value": "public",
            },
            Object {
              "kind": "whitespace",
              "value": " ",
            },
            Object {
              "kind": "keyword",
              "value": "function",
            },
            Object {
              "kind": "whitespace",
              "value": " ",
            },
            Object {
              "kind": "method",
              "value": "__construct",
            },
            Object {
              "kind": "plain",
              "value": "(",
            },
            Object {
              "kind": "keyword",
              "value": "private",
            },
            Object {
              "kind": "whitespace",
              "value": " ",
            },
            Object {
              "kind": "plain",
              "value": "$x",
            },
            Object {
              "kind": "plain",
              "value": ")",
            },
          ],
        },
        Object {
          "children": Array [],
          "endPosition": Point {
            "column": 17,
            "row": 14,
          },
          "representativeName": "Z",
          "startPosition": Point {
            "column": 17,
            "row": 14,
          },
          "tokenizedText": Array [
            Object {
              "kind": "keyword",
              "value": "abstract",
            },
            Object {
              "kind": "whitespace",
              "value": " ",
            },
            Object {
              "kind": "keyword",
              "value": "const",
            },
            Object {
              "kind": "whitespace",
              "value": " ",
            },
            Object {
              "kind": "method",
              "value": "Z",
            },
          ],
        },
        Object {
          "children": Array [],
          "endPosition": Point {
            "column": 13,
            "row": 15,
          },
          "representativeName": "X",
          "startPosition": Point {
            "column": 8,
            "row": 15,
          },
          "tokenizedText": Array [
            Object {
              "kind": "keyword",
              "value": "const",
            },
            Object {
              "kind": "whitespace",
              "value": " ",
            },
            Object {
              "kind": "method",
              "value": "X",
            },
          ],
        },
      ],
      "endPosition": Point {
        "column": 0,
        "row": 16,
      },
      "representativeName": "D",
      "startPosition": Point {
        "column": 15,
        "row": 9,
      },
      "tokenizedText": Array [
        Object {
          "kind": "keyword",
          "value": "abstract",
        },
        Object {
          "kind": "whitespace",
          "value": " ",
        },
        Object {
          "kind": "keyword",
          "value": "class",
        },
        Object {
          "kind": "whitespace",
          "value": " ",
        },
        Object {
          "kind": "class-name",
          "value": "D",
        },
      ],
    },
  ],
}
`;

exports[`outlineFromHackIdeOutline enum 1`] = `
Object {
  "outlineTrees": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [],
          "endPosition": Point {
            "column": 6,
            "row": 27,
          },
          "representativeName": "C",
          "startPosition": Point {
            "column": 2,
            "row": 27,
          },
          "tokenizedText": Array [
            Object {
              "kind": "keyword",
              "value": "const",
            },
            Object {
              "kind": "whitespace",
              "value": " ",
            },
            Object {
              "kind": "method",
              "value": "C",
            },
          ],
        },
      ],
      "endPosition": Point {
        "column": 0,
        "row": 28,
      },
      "representativeName": "En",
      "startPosition": Point {
        "column": 5,
        "row": 26,
      },
      "tokenizedText": Array [
        Object {
          "kind": "keyword",
          "value": "enum",
        },
        Object {
          "kind": "whitespace",
          "value": " ",
        },
        Object {
          "kind": "class-name",
          "value": "En",
        },
      ],
    },
  ],
}
`;

exports[`outlineFromHackIdeOutline function 1`] = `
Object {
  "outlineTrees": Array [
    Object {
      "children": Array [],
      "endPosition": Point {
        "column": 0,
        "row": 3,
      },
      "representativeName": "f",
      "startPosition": Point {
        "column": 9,
        "row": 2,
      },
      "tokenizedText": Array [
        Object {
          "kind": "keyword",
          "value": "function",
        },
        Object {
          "kind": "whitespace",
          "value": " ",
        },
        Object {
          "kind": "method",
          "value": "f",
        },
        Object {
          "kind": "plain",
          "value": "(",
        },
        Object {
          "kind": "plain",
          "value": "$x",
        },
        Object {
          "kind": "plain",
          "value": ",",
        },
        Object {
          "kind": "whitespace",
          "value": " ",
        },
        Object {
          "kind": "plain",
          "value": "$y",
        },
        Object {
          "kind": "plain",
          "value": ")",
        },
      ],
    },
  ],
}
`;

exports[`outlineFromHackIdeOutline interface 1`] = `
Object {
  "outlineTrees": Array [
    Object {
      "children": Array [],
      "endPosition": Point {
        "column": 13,
        "row": 24,
      },
      "representativeName": "I",
      "startPosition": Point {
        "column": 10,
        "row": 24,
      },
      "tokenizedText": Array [
        Object {
          "kind": "keyword",
          "value": "interface",
        },
        Object {
          "kind": "whitespace",
          "value": " ",
        },
        Object {
          "kind": "method",
          "value": "I",
        },
      ],
    },
  ],
}
`;

exports[`outlineFromHackIdeOutline trait 1`] = `
Object {
  "outlineTrees": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [],
          "endPosition": Point {
            "column": 2,
            "row": 21,
          },
          "representativeName": "f",
          "startPosition": Point {
            "column": 18,
            "row": 19,
          },
          "tokenizedText": Array [
            Object {
              "kind": "keyword",
              "value": "public",
            },
            Object {
              "kind": "whitespace",
              "value": " ",
            },
            Object {
              "kind": "keyword",
              "value": "function",
            },
            Object {
              "kind": "whitespace",
              "value": " ",
            },
            Object {
              "kind": "method",
              "value": "f",
            },
            Object {
              "kind": "plain",
              "value": "(",
            },
            Object {
              "kind": "plain",
              "value": ")",
            },
          ],
        },
      ],
      "endPosition": Point {
        "column": 0,
        "row": 22,
      },
      "representativeName": "T",
      "startPosition": Point {
        "column": 6,
        "row": 18,
      },
      "tokenizedText": Array [
        Object {
          "kind": "keyword",
          "value": "trait",
        },
        Object {
          "kind": "whitespace",
          "value": " ",
        },
        Object {
          "kind": "method",
          "value": "T",
        },
      ],
    },
  ],
}
`;
